src/skydb.ts

```diff
@@ -106,12 +106,28 @@ export async function setJSON(
     ...customOptions,
   };
 
+  // Create the data to upload to acquire its skylink.
+  const file = new File([JSON.stringify(json)], dataKey, { type: "application/json" });
+
+  // Start file upload, do not block.
+  const skyfilePromise: Promise<UploadRequestResponse> = this.uploadFile(file, opts);
+  let skylink: string;
+
   const privateKeyBuffer = Buffer.from(privateKey, "hex");
 
   if (revision === undefined) {
     // fetch the current value to find out the revision.
     const publicKey = pki.ed25519.publicKeyFromPrivateKey({ privateKey: privateKeyBuffer });
-    const entry: SignedRegistryEntry = await this.registry.getEntry(toHexString(publicKey), dataKey, opts);
+    // start getEntry, do not block.
+    const entryPromise: Promise<SignedRegistryEntry> = this.registry.getEntry(toHexString(publicKey), dataKey, opts);
+
+    // Block until both getEntry and Skyfile upload are finished.
+    const [entry, skyfile] = await Promise.all<SignedRegistryEntry, UploadRequestResponse>([
+      entryPromise,
+      skyfilePromise,
+    ]);
+    skylink = skyfile.skylink;
```

Comment:
I think we can move this line out of the `if` and `else` branches, it's not needed until later and would save 1 LoC.

Reply:
I was wondering why we're declaring the var at line 114 but I guess we can avoid that and just declare it with its init around L147?

Reply:
This may be me not understanding TS, but I get errors if I try to reference skyfile outside the if/else block since it was initialized inside it. This is the motivation for the weird `let skylink: string;` above.

Reply:
Oh, yes, your approach is absolutely correct - if you want to use the var in the block and then outside of it you need to declare it outside. But, as Marcin pointed out, you can just do it after the `if` block. That is, if I am reading the code correctly.

Reply:
https://www.typescriptlang.org/docs/handbook/variable-declarations.html#block-scoping

Reply:
I think what @ro-tex suggested with declaring `skyfile` before the scope should work. But it's such a minor thing that I'm just going to merge this :)

Reply:
Didn't see your post before putting the link there! But still, I get "Cannot find name: skyfile" because it was declared in the blockscope of the if statement.

Reply:
I'm too slow! I was missing an array restructuring assignment thing, but I'll do another pull. Thanks for the insight!